app/javascript/profileDropdown/flagButton.js

```diff
@@ -15,53 +17,56 @@ export function initFlag() {
     return;
   }
 
-  const user = userData();
-  if (!user) {
-    return;
-  }
+  getUserDataAndCsrfToken().then(({ _current_user, _csrfToken }) => {
+    const user = userData();
+    if (!user) {
+      flagButton.remove;
+      return;
+    }
 
-  const { profileUserId, profileUserName } = flagButton.dataset;
-  let isUserFlagged = flagButton.dataset.isUserFlagged === 'true';
-  const trustedOrAdmin = user.trusted || user.admin;
+    const { profileUserId, profileUserName } = flagButton.dataset;
+    let isUserFlagged = flagButton.dataset.isUserFlagged === 'true';
+    const trustedOrAdmin = user.trusted || user.admin;
 
-  if (!trustedOrAdmin || user.id === parseInt(profileUserId, 10)) {
-    flagButton.remove();
-  }
+    if (!trustedOrAdmin || user.id === parseInt(profileUserId, 10)) {
+      flagButton.remove();
+    }
 
-  function flag() {
-    const confirmFlag = window.confirm(
-      isUserFlagged
-        ? 'Are you sure you want to unflag this person? This will make all of their posts visible again.'
-        : 'Are you sure you want to flag this person? This will make all of their posts less visible.',
-    );
+    function flag() {
+      const confirmFlag = window.confirm(
+        isUserFlagged
+          ? 'Are you sure you want to unflag this person? This will make all of their posts visible again.'
+          : 'Are you sure you want to flag this person? This will make all of their posts less visible.',
+      );
 
-    if (confirmFlag) {
-      fetch('/reactions', {
-        method: 'POST',
-        headers: {
-          'X-CSRF-Token': window.csrfToken,
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({
-          reactable_type: 'User',
-          category: 'vomit',
-          reactable_id: profileUserId,
-        }),
-      })
-        .then((response) => response.json())
-        .then((response) => {
-          if (response.result === 'create') {
-            isUserFlagged = true;
-            flagButton.innerHTML = `Unflag ${profileUserName}`;
-          } else {
-            isUserFlagged = false;
-            flagButton.innerHTML = `Flag ${profileUserName}`;
-          }
+      if (confirmFlag) {
+        fetch('/reactions', {
+          method: 'POST',
+          headers: {
+            'X-CSRF-Token': window.csrfToken,
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            reactable_type: 'User',
+            category: 'vomit',
+            reactable_id: profileUserId,
+          }),
         })
-        .catch((e) => window.alert(`Something went wrong: ${e}`));
+          .then((response) => response.json())
```

Comment:
baby steps - this is the most JS I've dealt with in years :wink: 

but there's no time like the present to see what that looks like instead of `getPromise.then({...})` - thanks for the tip

Reply:
@djuber 😅 I feel you! I wrote most of the JS I have ever written after coming to Forem 😆